src/middleware/srgssr.js

```diff
@@ -193,6 +350,44 @@ class SrgSsr {
     );
   }
 
+  /**
+   * Get the current blocked segment from the player.
+   *
+   * @param {import('video.js/dist/types/player').default} player
+   *
+   * @returns {VTTCue|undefined} The blocked segment cue object, or undefined
+   */
+  static getBlockedSegment(player) {
+    const trackId = 'srgssr-blocked-segments';
+    const blockedSegmentsTrack = player.textTracks().getTrackById(trackId);
+
+    if (!blockedSegmentsTrack) return;
+
+    /** @type {[VTTCue]} */
+    const [blockedCue] = blockedSegmentsTrack.activeCues_;
+
+    return blockedCue;
+  }
+
+  /**
+   * Get the end time of a blocked segment.
+   *
+   * @param {import('video.js/dist/types/player').default} player
+   * @param {Number} currentTime
+   *
+   * @returns {Number|undefined} The end time of a blocked segment, or undefined
+   */
+  static getBlockedSegmentEndTime(player, currentTime) {
+    const blockedSegment = SrgSsr.getBlockedSegment(player);
```

Comment:
This is probably a corner case and most likely doesn't currently exist, but what would happen if there were 2 or more blocked segments at the same time? Check this case:

|    |  0 |  5 | 10 | 15 | 20 |
|----|----|----|----|----|----|
| Segment 1 |    | :blue_square:    |  :blue_square:  |    |    |
| Segment 2 |     |    |  :red_square:  | :red_square:   |    |

Shouldn't the end time be 15 ? Currently this function returns 10, and the player will seek twice to reach the end of the second blocked segment.